app/services/service_provider_updater.rb

```diff
@@ -20,11 +20,24 @@ def update_local_caches(service_provider)
 
   def update_cache(issuer, service_provider)
     if service_provider['active'] == true
-      ServiceProvider.find_or_create_by!(issuer: issuer) do |sp|
-        sp.attributes = cleaned_service_provider(service_provider)
-      end
+      create_or_update_service_provider(issuer, service_provider)
     else
-      ServiceProvider.destroy_all(issuer: issuer, approved: false)
+      ServiceProvider.destroy_all(issuer: issuer, native: false)
+    end
+  end
+
+  def create_or_update_service_provider(issuer, service_provider)
+    sp = ServiceProvider.from_issuer(issuer)
+    return if sp.native?
+    sync_model(sp, cleaned_service_provider(service_provider))
+  end
+
+  def sync_model(sp, cleaned_attributes)
+    if sp.is_a?(NullServiceProvider)
```

Comment:
What if `from_issuer` returned `ServiceProvider.new(issuer: issuer)` when none is found rather than `NullServiceProvider`? That way, we could set the attributes and save the same way whether `from_issuer` returns an existing or new record...